src/Workspaces/Core/Portable/FlowAnalysis/SymbolUsageAnalysis/SymbolUsageAnalysis.DataFlowAnalyzer.FlowGraphAnalysisData.cs

```diff
@@ -195,6 +189,100 @@ private PooledHashSet<(ISymbol, IOperation)> GetOrCreateSymbolWritesInBlockRange
                     return writesInBlockRange;
                 }
 
+                private void AddOperationsInRange(
+                    ControlFlowGraph cfg,
+                    int firstBlockOrdinal,
+                    int lastBlockOrdinal,
+                    PooledHashSet<IOperation> operationsBuilder)
+                {
+                    // Compute all descendant operations in basic block range.
+                    for (int i = firstBlockOrdinal; i <= lastBlockOrdinal; i++)
+                    {
+                        foreach (var operation in cfg.Blocks[i].DescendantOperations())
+                        {
+                            var added = operationsBuilder.Add(operation);
+                            if (added && operation is IInvocationOperation invocation)
+                            {
+                                if (invocation.Instance != null &&
+                                    _reachingDelegateCreationTargets.TryGetValue(invocation.Instance, out var targets))
+                                {
+                                    AddOperationsFromDelegateCreationTargets(targets);
+                                }
+                                else if (invocation.TargetMethod.IsLocalFunction())
+                                {
+                                    var localFunctionGraph = cfg.GetLocalFunctionControlFlowGraphInScope(invocation.TargetMethod);
+                                    if (localFunctionGraph != null)
+                                    {
+                                        AddOperationsInLambdaOrLocalFunctionGraph(localFunctionGraph);
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    return;
+
+                    // Local functions.
+                    void AddOperationsFromDelegateCreationTargets(PooledHashSet<IOperation> targets)
+                    {
+                        foreach (var target in targets)
+                        {
```

Comment:
can this be.. like GetDecedentControlFlowGraph API? it would be hard for anyone except you would know what all controlflowgraph they need to take care for all sub tree operations..

Reply:
Sure, I will rename it.